src/hooks/vaultsHooks.ts

```diff
@@ -261,43 +154,64 @@ export const useLoadedVaultPositions = () => {
     queryKey: ['vaultPositions'],
     queryFn: async () => {
       return wrapNullable(
-        VaultCalculator.getVaultPositionsResponse(await placeholderFetchMegavaultPositions())
+        VaultCalculator.getVaultPositionsResponse(
+          safeStringifyForAbacus(await getMegavaultPositions())
+        )
       );
     },
     ...vaultQueryOptions,
+    refetchInterval: timeUnits.second * 15,
   });
 
   const calculatedPositions = useMemo(() => {
-    if (vaultPositions?.data == null || marketsMap == null || marketsMap.size === 0) {
+    if (vaultPositions?.data == null || marketsMap.data == null || marketsMap.data.size === 0) {
       return undefined;
     }
     return VaultCalculator.calculateVaultPositions(
       vaultPositions.data,
       subvaultHistories?.data,
-      marketsMap
+      marketsMap.data
     );
   }, [subvaultHistories, vaultPositions, marketsMap]);
 
   return calculatedPositions;
 };
 
-const vaultAccountQueryKey = ['vaultAccount'];
-
-export function forceRefreshVaultAccount() {
-  appQueryClient.invalidateQueries({ queryKey: vaultAccountQueryKey, exact: true });
+export function useForceRefreshVaultAccount() {
+  const { dydxAddress } = useAccounts();
+  return useCallback(
+    () =>
+      appQueryClient.invalidateQueries({
+        queryKey: ['vaultAccount', dydxAddress, true],
+        exact: true,
+      }),
+    [dydxAddress]
+  );
 }
 
 export const useLoadedVaultAccount = () => {
+  const { getAllAccountTransfersBetween } = useDydxClient();
+  const { dydxAddress } = useAccounts();
+  const { getVaultAccountInfo } = useSubaccount();
+  const { compositeClient } = useDydxClient();
```

Comment:
`useDydxClient` is used twice. destructure from one call?

Reply:
oops